### ISSUE TITLE
[Accessibility] `aria-readonly` attribute should be set on readonly row cells

### ISSUE THREAD

**Issue by matsbryntse** [bryntum/support#9257]
[Forum post](https://forum.bryntum.com/viewtopic.php?f=51&t=29195&p=147175#p147175)

Hi,

We have a grid where some of the records are editable and some are read-only depending on different conditions.

Currently, we are listening to the Grid.feature.CellEdit#event-beforeCellEditStart event and returning false for the record when editing should be prevented.

Whilst this works, we are concerned that we aren't implementing this feature in the desired way. Our concern comes from that controlling the editable behaviour in this manner means that the "readonly" property in the accessibility tree isn't updated correctly, resulting in our screen reader users have no idea whether the cell editable or read only (see screenshot).

![Screenshot 2024-05-28 at 15.36.39.png](https://forum.bryntum.com/download/file.php?id=28663)

To get around this, we tried setting Core.data.Model#field-readOnly to true on the model class. However, this does not seem to prevent the widget inside the cell from opening nor does it set "readonly" to true in the accessibility tree.

Please can you advise on the following:
Are we implementing our requirement correctly?
Is the "readonly" property not being set in the accessibility tree a bug?

Thanks!
![Screenshot 2024-05-28 at 15.36.39.png](https://forum.bryntum.com/download/file.php?id=28663)
